rotkehlchen/globaldb/handler.py

```diff
@@ -258,16 +259,17 @@ def retrieve_assets(filter_query: 'AssetsFilterQuery') -> Tuple[List[Dict[str, A
         """  # noqa: E501
         query = f'SELECT * FROM ({parent_query}) {prepared_filter_query}'
         underlying_tokens_query = """
-        SELECT assets.identifier AS identifier, address, token_kind, parent_token_entry, weight, name, symbol, type, chain
+        SELECT parent_token_entry AS identifier, underlying_tokens_list.identifier, token_kind, weight, name, symbol, type, chain
         FROM underlying_tokens_list
-        INNER JOIN assets ON assets.identifier = parent_token_entry
+        LEFT JOIN assets ON assets.identifier = parent_token_entry
         INNER JOIN evm_tokens ON assets.identifier=evm_tokens.identifier
         INNER JOIN common_asset_details ON common_asset_details.identifier = assets.identifier
         """  # noqa: E501
         with GlobalDBHandler().conn.read_ctx() as cursor:
             # get all underlying tokens
             for entry in cursor.execute(f'SELECT * FROM ({underlying_tokens_query}) {prepared_filter_query}', bindings):  # noqa: E501
-                underlying_tokens[entry[0]].append(UnderlyingToken.deserialize_from_db((entry[1], entry[2], entry[4])).serialize())  # noqa: E501
+                address = caips_identifier_to_address(entry[1])
```

Comment:
Isn't it better to query address from the database?

Reply:
Rewrote this section improving it after reviewing a bug with kelsos